Source/DafnyCore/Backends/Dafny/DafnyCodeGenerator.cs

```diff
@@ -1223,41 +1283,45 @@ private class ExprLvalue : ILvalue {
         }
 
         builder.Builder.AddExpr(baseExpr);
+      } else if (GetStatementBuilder(wr, out var builder2)) {
+        builder2.Builder.AddStatement(StatementContainer.UnsupportedToStatement($"Cannot emit literal expression {e} outside of expression context: " + wr.GetType()));
```

Comment:
Is there a mechanism to scan for these unsupporteds and fail the pipeline if any are present?

Reply:
With the new option `--emit-uncompilable-code`, doing so would be superfluous. I added a check here as well that the option --emit-uncompilable-code must be present for this to occur.